legend-engine-language-pure-dsl-service-execution/src/test/java/org/finos/legend/engine/language/pure/dsl/service/execution/TestServiceRunner.java

```diff
@@ -376,6 +376,38 @@ public void SimpleM2MServiceRunnerForRecursiveExecution()
         Assert.assertEquals("{\"wheels\":55}", result);
     }
 
+    private static class SimpleM2MServiceRunnerForDateTimeSerialization extends AbstractServicePlanExecutor
+    {
+        SimpleM2MServiceRunnerForDateTimeSerialization(String function)
+        {
+            super("test::Service", buildPlanForFetchFunction("/org/finos/legend/engine/pure/dsl/service/execution/test/simpleM2MService.pure", function), true);
+        }
+
+        @Override
+        public List<ServiceVariable> getServiceVariables()
+        {
+            return Collections.emptyList();
+        }
+    }
+
+    @Test
+    public void SimpleM2MServiceRunnerForDateTimeSerialization()
+    {
+        SimpleM2MServiceRunnerForDateTimeSerialization validDateTimeSerialization = new SimpleM2MServiceRunnerForDateTimeSerialization("test::serializeDateTime__String_1_");
+        ServiceRunnerInput serviceRunnerInput1 = ServiceRunnerInput
+                .newInstance()
+                .withSerializationFormat(SerializationFormat.PURE);
+        String result = validDateTimeSerialization.run(serviceRunnerInput1);
+        Assert.assertEquals("{\"age\":25,\"birthDate\":\"2014-02-27T15:01:35.231+0000\"}", result);
+
+        SimpleM2MServiceRunnerForDateTimeSerialization invalidDateTimeSerialization = new SimpleM2MServiceRunnerForDateTimeSerialization("test::serializeInvalidDateTimeFormat__String_1_");
+        ServiceRunnerInput serviceRunnerInput2 = ServiceRunnerInput
+                .newInstance();
+        IllegalArgumentException e1 = Assert.assertThrows(IllegalArgumentException.class, () -> invalidDateTimeSerialization.run(serviceRunnerInput2));
+        MatcherAssert.assertThat(e1, ThrowableMessageMatcher.hasMessage(CoreMatchers.startsWith("java.lang.IllegalArgumentException: Too many pattern letters: m\n" +
+                "yyyy-MM-dd\"T\"HH:mmm:ss.SSSZ is not a valid dateTime format in SerializationConfig")));
```

Comment:
is it possible to inform users of available formats?

Reply:
Each symbol used in DateTimeFormatter patterns have different set of possible patterns and so they can be combined as per requirement. The possible list would be too long and the exception that users will encounter will point out what symbol in the pattern is problematic.

Reply:
What are these rules/possibilities?  Normal JDK?  If that is the case, wonder if we should include a link to the JDK docs (?)

Reply:
We will list out the possibilities on studio